src/main/java/net/dv8tion/jda/api/entities/Entitlement.java

```diff
@@ -137,11 +140,69 @@ default String getGuildId()
     @Nullable
     OffsetDateTime getTimeEnding();
 
+    /**
+     * The payment data for this {@link Entitlement Entitlement}
+     *
+     * @return The payment data for this {@link Entitlement Entitlement}
+     */
+    @Nullable
+    PaymentData getPaymentData();
+
+    /**
+     * Whether the {@link Entitlement Entitlement} was consumed or not.
+     *
+     * @return True if the {@link Entitlement Entitlement} was consumed, False otherwise
+     */
+    boolean wasConsumed();
+
+    /**
+     * Consumes the {@link Entitlement Entitlement} if it has not already been consumed.
+     *
+     * @throws IllegalStateException
+     *        If the {@link Entitlement Entitlement} has already been consumed
+     *
+     * @return A {@link RestAction} that will consume the {@link Entitlement Entitlement}
+     */
+    @Nonnull
+    @CheckReturnValue
+    RestAction<Void> consume();
+
     /**
      * Represents the type of this Entitlement
      */
     enum EntitlementType
     {
+        /**
+         * Entitlement was purchased by user
+         */
+        PURCHASE(1),
```

Comment:
None of these are in the spec, are these actually observed for bots?

Reply:
i tried it myself and the other types seem to work